src/coreclr/jit/lower.cpp

```diff
@@ -1775,14 +1781,67 @@ GenTree* Lowering::AddrGen(void* addr)
     return AddrGen((ssize_t)addr);
 }
 
+//------------------------------------------------------------------------
+// LowerCallMemmove: Replace Buffer.Memmove(DST, SRC, CNS_SIZE) with a GT_STORE_BLK:
+//
+//    *  STORE_BLK struct<CNS_SIZE> (copy) (Unroll)
+//    +--*  LCL_VAR   byref  dst
+//    \--*  BLK       struct
+//       \--*  LCL_VAR   byref  src
+//
+// Arguments:
+//    tree - GenTreeCall node to replace with STORE_BLK
+//
+GenTree* Lowering::LowerCallMemmove(GenTreeCall* call)
+{
+    assert(comp->lookupNamedIntrinsic(call->gtCallMethHnd) == NI_System_Buffer_Memmove);
+    assert(call->gtArgs.CountArgs() == 3);
+
+    GenTree* lengthArg = call->gtArgs.GetArgByIndex(2)->GetNode();
+    if (lengthArg->IsIntegralConst())
+    {
+        ssize_t cnsSize = lengthArg->AsIntCon()->IconValue();
+        // TODO-CQ: drop the whole thing in case of 0
+        if ((cnsSize > 0) && (cnsSize <= (ssize_t)comp->getUnrollThreshold(Compiler::UnrollKind::Memmove)))
+        {
+            GenTree* dstOp = call->gtArgs.GetArgByIndex(0)->GetNode();
+            GenTree* srcOp = call->gtArgs.GetArgByIndex(1)->GetNode();
+
+            // TODO-CQ: Try to create an addressing mode
+            GenTreeBlk* srcBlk = comp->gtNewBlockVal(srcOp, (unsigned)cnsSize)->AsBlk();
+            srcBlk->ChangeOper(GT_IND);
+            srcBlk->SetContained();
+
+            GenTreeBlk* dstBlk = comp->gtNewBlockVal(dstOp, (unsigned)cnsSize)->AsBlk();
+            dstBlk->SetOperRaw(GT_STORE_BLK);
+            dstBlk->gtFlags |= (GTF_BLK_UNALIGNED | GTF_IND_ASG_LHS | GTF_ASG | GTF_EXCEPT);
+            dstBlk->AsBlk()->Data() = srcBlk;
```

Comment:
```suggestion
            GenTreeBlk* storeBlk = new (comp, GT_STORE_BLK)
                GenTreeBlk(GT_STORE_BLK, TYP_STRUCT, dstAddr, srcBlk, comp->typGetBlkLayout(cnsSize));
            storeBlk->gtFlags |= (GTF_BLK_UNALIGNED | GTF_ASG | GTF_EXCEPT | GTF_GLOB_REF);
```

`gtNewBlockVal` can return a local variable so this is not correct as-is.